src/librustc_trans/back/linker.rs

```diff
@@ -198,8 +201,30 @@ impl<'a> Linker for GnuLinker<'a> {
         self.cmd.arg("-Wl,-Bdynamic");
     }
 
-    fn export_symbols(&mut self, _: &Session, _: &CrateTranslation, _: &Path) {
-        // noop, visibility in object files takes care of this
+    fn export_symbols(&mut self,
+                      sess: &Session,
+                      trans: &CrateTranslation,
+                      tmpdir: &Path,
+                      crate_type: CrateType) {
+        let path = tmpdir.join("list");
+        let res = (|| -> io::Result<()> {
+            let mut f = BufWriter::new(File::create(&path)?);
+            for sym in exported_symbols(sess, trans, crate_type) {
+                writeln!(f, "  {}", sym)?;
```

Comment:
This doesn't seem to be enough for iOS, [see failure](http://buildbot.rust-lang.org/builders/auto-mac-cross-ios-opt/builds/279/steps/compile/logs/stdio).